back/db/migrate/20210430154637_add_confirmation_required_field_to_users.user_confirmation.rb

```diff
@@ -0,0 +1,17 @@
+class AddConfirmationRequiredFieldToUsers < ActiveRecord::Migration[6.0]
+  def up
+    add_column :users, :confirmation_required, :boolean, default: true, null: false
+    confirm_active_users
+  end
+
+  def down
+    remove_column :users, :confirmation_required, :boolean, default: true, null: false
+  end
+
+  def confirm_active_users
+    active_users = User.active
+    user_ids = active_users.reject(&:should_require_confirmation?).map(&:id)
```

Comment:
This is still processing all users in ruby 😅

Reply:
Ah I thought you meant the amount of SQL queries, now its reduced to 2. The problem here is that, to check phone numbers, I'd have to write a special query to check if the user signed up with a phone and another one for the highest role. I don't really see a nice way of doing that. Any suggestions?

Reply:
Then I'm not sure if we can accept this, I would like to have @kogre confirmation that this is acceptable.

Perhaps we could find a compromise? E.g doing `User.active.pluck :email` to get a list of emails (thereby avoiding loading all users into memory) and check for all of these emails if they are not phone numbers? I think we can assume that emails are unique for users so then you could do `User.normal_user.where.not(email: phone_numbers).update_all(confirmation_required: false)`?

Reply:
I don't think loading users should be an issue since migrations run separately for each tenant, thus loading only one tenant's users at a time. It's something we do across our app a lot already, is there any caveat with migrations that I'm not seeing?

Otherwise your solution sounds good as well

Reply:
It's true that we're occasionally doing that with some rake tasks, but those also take a long time to execute. In the case of a migration, the whole application might not work properly as long as the migrations are running. 

Also, the code is depending on the existence of e.g. the `should_require_confirmation?` method. I'm not sure how big a deal that is, but I know Koen really prefers to avoid these dependencies and instead try to do everything purely in SQL.

Would it be an option to instead not do this in the migration at all, but run separate rake tasks that set these values after the release? Then for me everything is good to go :)

Reply:
Good point, I think thats a better option indeed. Ill move the data migration into a rake task 👍